testutils/generators.go

```diff
@@ -23,6 +23,130 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+func GenerateRequestRecord(pulse insolar.PulseNumber, objectID insolar.ID) *exporter.Record {
+	r := GenerateRecordsSilence(1)[0]
+	id := gen.IDWithPulse(pulse)
+	r.Record.ID = id
+	r.Record.ObjectID = objectID
+	r.ShouldIterateFrom = nil
+	reference := insolar.NewReference(id)
+	r.Record.Virtual.Union = &insrecord.Virtual_IncomingRequest{
+		IncomingRequest: &insrecord.IncomingRequest{
+			Object: reference,
+		},
+	}
+	return r
+}
+
+func GenerateVirtualActivateRecord(pulse insolar.PulseNumber, objectID, requestID insolar.ID) (record *exporter.Record) {
+	r := GenerateRecordsSilence(1)[0]
+	id := gen.IDWithPulse(pulse)
+	r.Record.ID = id
+	r.Record.ObjectID = objectID
+	r.ShouldIterateFrom = nil
+	requestRerence := insolar.NewReference(requestID)
+	r.Record.Virtual.Union = &insrecord.Virtual_Activate{
+		Activate: &insrecord.Activate{
+			Image:   gen.Reference(),
+			Request: *requestRerence,
+		},
+	}
+	return r
+}
+
+func GenerateVirtualAmendRecordsLinkedArray(pulse insolar.PulseNumber, jetID insolar.JetID, objectID, prevStateID insolar.ID, recordsCount int) []*exporter.Record {
+	result := make([]*exporter.Record, recordsCount)
+	for i := 0; i < recordsCount; i++ {
+		r := GenerateVirtualAmendRecord(pulse, objectID, prevStateID)
+		r.Record.JetID = jetID
+		result[i] = r
+		prevStateID = r.Record.ID
+	}
+	return result
+}
+
+func GenerateVirtualAmendRecord(pulse insolar.PulseNumber, objectID, prevStateID insolar.ID) *exporter.Record {
+	r := GenerateRecordsSilence(1)[0]
+	id := gen.IDWithPulse(pulse)
+	r.Record.ID = id
+	r.Record.ObjectID = objectID
+	r.ShouldIterateFrom = nil
+	r.Record.Virtual.Union = &insrecord.Virtual_Amend{
+		Amend: &insrecord.Amend{
+			Image:     gen.Reference(),
+			PrevState: prevStateID,
+		},
+	}
+	return r
+}
+
+func GenerateVirtualDeactivateRecord(pulse insolar.PulseNumber, objectID, prevStateID insolar.ID) *exporter.Record {
+	r := GenerateRecordsSilence(1)[0]
+	id := gen.IDWithPulse(pulse)
+	r.Record.ID = id
+	r.Record.ObjectID = objectID
+	r.ShouldIterateFrom = nil
+	r.Record.Virtual.Union = &insrecord.Virtual_Deactivate{
+		Deactivate: &insrecord.Deactivate{
+			PrevState: prevStateID,
+		},
+	}
+	return r
+}
+
+type ObjectLifeline struct {
+	States []StateByPulse
+	ObjID  insolar.ID
+}
+
+type StateByPulse struct {
+	Pn      insolar.PulseNumber
+	Records []*exporter.Record
+}
+
+func GenerateObjectLifeline(pulsesNumber, recordsInPulse int) ObjectLifeline {
```

Comment:
lets rename `pulsesNumber` -> `pulseCount`, because now it seems like we pass specific PulsesNumber field for all records